### PR TITLE
VirtualEnv can be created from setup.py files

### LINKED ISSUE #275: Support requirements declared via setup.py
Currently Component Specs allow a requirements.txt to be specified via the "requirements_file" spec field. Many of [the RL frameworks](https://docs.google.com/spreadsheets/d/1r68IytgMKmZ7Se0teY0LlRRE7YLzflu4v1LuO-UK1Vs/edit#gid=265043462) we have encountered specify their requirements via a `setup.py` instead of a `requirements.txt` file. We should support both.

For example, the following projcects have a setup.py:

- https://github.com/rail-berkeley/rlkit (Ashvin & VItchyr's)
- https://github.com/ikostrikov/pytorch-a2c-ppo-acktr-gail (Ilya's repo)
- https://github.com/facebookresearch/ReAgent
- https://github.com/DLR-RM/stable-baselines3
- https://github.com/takuseno/d3rlpy -- has both setup.py and requirements.txt

### PR DESCRIPTION
**Pull request by nickjalbert**
This PR  allows virtual environments to be created based on the `install_requires` kwarg found in `setup.py` files.  Also included in this PR:

* More cache-clearing commands:
    * `agentos clear-env-cache`
    * `agentos clear-repo-cache`
    * `agentos clear-cache  # clears both env and repo cache`

* You can list multiple requirements files (Python package `setup.py` files or pip-compatible `requirements.txt` files) in the `requirements_path` Component spec key separated by a semicolon and all will be installed into the virtual env.
* Update to `rllib_agent` and `sb3_agent` requirements.
* Minor improvements to the `install_requirements.py` environment setup script.
* Remove unneeded requirements from AgentOS `setup.py`.

Try the following demo in your Python REPL to get a Policy class component out of Ilya's repo:

```python
from agentos import Component, Repo
ilya_repo = Repo.from_github("ikostrikov", "pytorch-a2c-ppo-acktr-gail")
ilya_policy = Component.from_repo( repo=ilya_repo, identifier="Policy==41332b78dfb50321c29bade65f9d244387f68a60", class_name="Policy", file_path="./a2c_ppo_acktr/model.py", requirements_path="./setup.py", instantiate=False)
policy_obj = ilya_policy.get_object()
```

Fixes #275.

### CODE REVIEW

File: agentos/component.py [hunk 1]
```diff
@@ -440,10 +440,11 @@ def _build_virtual_env(self) -> VirtualEnv:
...
             if c.requirements_path is None:
                 continue
-            full_req_path = self.repo.get_local_file_path(
-                c.identifier.version, c.requirements_path
-            ).absolute()
-            req_paths.add(full_req_path)
+            for req_path in c.requirements_path.split(";"):
+                full_req_path = self.repo.get_local_file_path(
+                    c.identifier.version, req_path
+                ).absolute()
```
Comment (andyk) [inline agentos/component.py:446]: Would it make sense to assert that the requirements_file exists here?

**Review by andyk — APPROVED** [review]
LGTM! Just 3 nits/typos in docstrings and one possible suggestion for an assert statement.

**Comment by andyk** [comment]
@nick I think this might close #297?

**Comment by nickjalbert** [comment]
> I think this might fix https://github.com/agentos-project/agentos/issues/297?

I don't think this goes quite as far as I'm hoping to in #297, I think we should keep that open for now...

**Pull request merged**